File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/ArtistProfileDto.java
package com.example.miraculousbackend.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ArtistProfileDto {

    private Long id;
    private String login;
    private String dateOfRegistration;
    private List<String> genres;
    private String description;
    private String location;
    private String pictureUrlName;
    private Long totalNumberOfLikes;
    private Boolean areCommentsAllowed;
}

File: miraculous-frontend/miraculous-ui/src/dto/artist-profile-dto/artist-profile-dto.ts
export class ArtistProfileDto {

  id: number;
  login: string;
  dateOfRegistration: string;
  genres: string[];
  description: string;
  location: string;
  pictureUrlName: string;
  totalNumberOfLikes: number;
  areCommentsAllowed: boolean;


  constructor(id?: number, login?: string, dateOfRegistration?: string, genres?: string[], description?:
    string, location?: string, pictureUrlName?: string, totalNumberOfLikes?: number, areCommentsAllowed?: boolean) {
    this.id = id;
    this.login = login;
    this.dateOfRegistration = dateOfRegistration;
    this.genres = genres;
    this.description = description;
    this.location = location;
    this.pictureUrlName = pictureUrlName;
    this.totalNumberOfLikes = totalNumberOfLikes;
    this.areCommentsAllowed = areCommentsAllowed;
  }
}
